analytics/terraform/superset-on-eks/addons.tf

```diff
@@ -0,0 +1,166 @@
+
+#---------------------------------------------------------------
+# IRSA for EBS CSI Driver
+#---------------------------------------------------------------
+module "ebs_csi_driver_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.20"
+  role_name_prefix      = format("%s-%s", local.name, "ebs-csi-driver-")
+  attach_ebs_csi_policy = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+  tags = local.tags
+}
+
+resource "kubernetes_ingress_class_v1" "aws-alb" {
+  metadata {
+    name = "aws-alb"
+  }
+
+  spec {
+    controller = "ingress.k8s.aws/alb"
+  }
+}
+
+module "lb_role" {
+ source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+ role_name                              = format("%s-%s", local.name, "lb-controller-role")
+ attach_load_balancer_controller_policy = true
+
+ oidc_providers = {
+     main = {
+     provider_arn               =  module.eks.oidc_provider_arn
+     namespace_service_accounts = ["kube-system:aws-load-balancer-controller"]
+     }
+ }
+ }
+
+
+resource "kubernetes_service_account" "service-account" {
+ metadata {
+     name      = "aws-load-balancer-controller"
+     namespace = "kube-system"
+     labels = {
+     "app.kubernetes.io/name"      = "aws-load-balancer-controller"
+     "app.kubernetes.io/component" = "controller"
+     }
+     annotations = {
+     "eks.amazonaws.com/role-arn"               = module.lb_role.iam_role_arn
+     "eks.amazonaws.com/sts-regional-endpoints" = "true"
+     }
+ }
+ depends_on = [ module.eks ]
+ }
+
+ 
+ resource "helm_release" "alb-controller" {
```

Comment:
This addon is available under EKS Addons. Checkout this link https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L128C1-L131C4